ql/src/java/org/apache/hadoop/hive/ql/security/authorization/HiveCustomStorageHandlerUtils.java

```diff
@@ -71,4 +73,26 @@ public static void setWriteOperation(Configuration conf, String tableName, Conte
 
         conf.set(WRITE_OPERATION_CONFIG_PREFIX + tableName, operation.name());
     }
+
+    public static boolean isWriteOperation(Configuration conf, String tableName) {
```

Comment:
is this they way we try to identify operation type, it looks not very straitforward

Reply:
the conf used is different, that is used by FileSink, If I play with that the projectSchema will consume it and leads to different world.
So, created a different for FileScan.